robot-detect

```diff
@@ -34,20 +34,40 @@ MSG_FASTOPEN = 0x20000000
 EXECUTE_BLINDING = True
 
 
-def get_rsa_from_server(server, port):
+def create_socket(ip_family=4, timeout=None):
+    if ip_family == 4:
+        s = socket.socket()
+        if timeout is not None:
+            s.settimeout(timeout)
+        return s
+    s = socket.socket(socket.AF_INET6)
```

Comment:
Why is a different kind of socket required?

Reply:
`AF_INET` seems to be IPv4-only.

From https://docs.python.org/3/library/socket.html#socket-families, on `AF_INET`:

> This behavior is not compatible with IPv6, therefore, you may want to avoid these if you intend to support IPv6 with your Python programs.

Reply:
I see.  But I'm not sure we need that here.  You could pass the parameters that you compute the very first time with `getaddrinfo` or `create_connection`/`getpeername`/`getsockname` to that function, and then pass them further to `socket.socket` and `socket.connect`.  This way you don't need this extra `if-then-else`.

Reply:
Unfortunately, in the `oracle` function I need to create a `socket` but not a connection, in order to allow the `fastopen` if available. However I think I have managed to solve this issue in a polished way (line 193). In this case it is simply not possible to bypass the choice of IP type, since the `sendto` function used in case `fastopen` is available should not work on an open connection.